libcudacxx/test/libcudacxx/std/utilities/meta/meta.unary/meta.unary.prop/is_scoped_enum.pass.cpp

```diff
@@ -0,0 +1,151 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// <cuda/std/type_traits>
+
+// is_scoped_enum
+
+#include <cuda/std/cstddef> // for cuda::std::nullptr_t
+#include <cuda/std/type_traits>
+
+#include "test_macros.h"
+
+template <class T>
+__host__ __device__ void test_positive()
+{
+  static_assert(cuda::std::is_scoped_enum<T>::value, "");
+  static_assert(cuda::std::is_scoped_enum<const T>::value, "");
+  static_assert(cuda::std::is_scoped_enum<volatile T>::value, "");
+  static_assert(cuda::std::is_scoped_enum<const volatile T>::value, "");
+
+#if !defined(_CCCL_NO_VARIABLE_TEMPLATES)
+  static_assert(cuda::std::is_scoped_enum_v<T>, "");
+  static_assert(cuda::std::is_scoped_enum_v<const T>, "");
+  static_assert(cuda::std::is_scoped_enum_v<volatile T>, "");
+  static_assert(cuda::std::is_scoped_enum_v<const volatile T>, "");
+#endif // !_CCCL_NO_VARIABLE_TEMPLATES
+}
+
+template <class T>
+__host__ __device__ void test_negative()
+{
+  static_assert(!cuda::std::is_scoped_enum<T>::value, "");
+  static_assert(!cuda::std::is_scoped_enum<const T>::value, "");
+  static_assert(!cuda::std::is_scoped_enum<volatile T>::value, "");
+  static_assert(!cuda::std::is_scoped_enum<const volatile T>::value, "");
+
+#if !defined(_CCCL_NO_VARIABLE_TEMPLATES)
+  static_assert(!cuda::std::is_scoped_enum_v<T>, "");
+  static_assert(!cuda::std::is_scoped_enum_v<const T>, "");
+  static_assert(!cuda::std::is_scoped_enum_v<volatile T>, "");
+  static_assert(!cuda::std::is_scoped_enum_v<const volatile T>, "");
+#endif // !_CCCL_NO_VARIABLE_TEMPLATES
+}
+
+class Empty
+{};
+
+class NotEmpty
+{
+  virtual ~NotEmpty();
+};
+
+union Union
+{};
+
+struct bit_zero
+{
+  int : 0;
+};
+
+class Abstract
+{
+  virtual ~Abstract() = 0;
+};
+
+enum Enum
+{
+  zero,
+  one
+};
+enum class CEnum1
+{
+  zero,
+  one
+};
+enum class CEnum2;
+enum class CEnum3 : short;
+struct incomplete_type;
+
+using FunctionPtr  = void (*)();
+using FunctionType = void();
+
+struct TestMembers
+{
+  static int static_method(int)
```

Comment:
Can you add `__host__ __device__` here and all applicable other places to appease NVRTC?

Reply:
I thought it would be fine as only the type is used, not the actual function.